### PR TITLE
chore: Remove miscellaneous deprecated exports & prototypes

### PR DESCRIPTION
**Pull request by jimblanc**
#### Description of changes
This change removes a variety of deprecated exports & functional prototypes ahead of the breaking v5 release.

Specifically:
- Removes the `JS` object from `core` in favor of individual functional exports
- Removes a deprecated `Analytics.record` function prototype
  - This function has been deprecated since `aws-amplify@4.3.15`: https://github.com/aws-amplify/amplify-js/releases/tag/aws-amplify%404.3.15
- Removes two additional default exports that were missed

#### Description of how you validated changes
Unit tests, checked integ tests & sample apps for usage of removed exports. Cleanish CircleCI run (failures not due to this change).

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/core/__tests__/JS-browser-runtime-test.ts [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
+
+/** The doc block above is to change the running environment of Jest to
+ * jsdom (which is also the default) Since this is allowed per test file
+ * and not per test or describe, we have two tests, one for node and other for browser
+ */
+import * as core from '../dist/aws-amplify-core.js';
+
+describe('JS browserOrNode build test', () => {
+	test('when its browser ', () => {
+		expect(core.browserOrNode()).toStrictEqual({
```
Comment (cshfang) [inline packages/core/__tests__/JS-browser-runtime-test.ts:13]: Nit: Could we just import what we need?
```suggestion
import { browserOrNode } from '../dist/aws-amplify-core.js';

describe('browserOrNode build test', () => {
	test('when it\'s browser ', () => {
		expect(browserOrNode()).toStrictEqual({
```

File: packages/core/__tests__/JS-node-runtime-test.ts [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
+
+/** The doc block above is to change the running environment of Jest to node.
+ * Since this is allowed per test file and not per test or describe, we have
+ * two tests, one for node and other for browser
+ */
+import * as core from '../lib';
+
+describe('JS build test', () => {
+	test('when its node ', () => {
+		expect(core.browserOrNode()).toStrictEqual({
```
Comment (cshfang) [inline packages/core/__tests__/JS-node-runtime-test.ts:13]: Nit: Could we just import what we need?
```suggestion
import { browserOrNode } from '../lib';

describe('browserOrNode build test', () => {
	test('when it\'s node ', () => {
		expect(browserOrNode()).toStrictEqual({
```

**Review by cshfang — APPROVED** [review]
Non blocking suggestion. Lgtm!

**Review by AllanZhengYP — APPROVED** [review]

**Review by elorzafe — COMMENTED** [review]
Can we remove unused code for Analytics change or we can also do it in a separate PR, whatever works @jimblanc 

Thanks!

File: packages/analytics/src/Analytics.ts [hunk 3]
```diff
@@ -256,19 +256,6 @@ export class AnalyticsClass {
...
-	 * @return - A promise which resolves if buffer doesn't overflow
-	 */
-	public async record(
-		eventName: string,
-		attributes?: EventAttributes,
-		metrics?: EventMetrics
-	);
 	public async record(
 		event:
 			| string
```
[the remark this answers is not in the document]
Comment (jimblanc) [reply packages/analytics/src/Analytics.ts]: Good catch, cleaned this up more fully in the latest diff

File: packages/core/__tests__/JS-node-runtime-test.ts (hunk 2 above)
Comment (jimblanc) [reply packages/core/__tests__/JS-node-runtime-test.ts]: Updated in latest diff

**Review by elorzafe — APPROVED** [review]
Looks good to me

Thanks @jimblanc 🎉

**Pull request merged**